sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.students.controller.ts

```diff
@@ -34,12 +36,38 @@ export class RestrictionStudentsController extends BaseController {
         studentToken.studentId,
         {
           onlyActive: true,
+          filterNoEffectRestrictions: true,
         },
       );
-
-    return studentRestrictions.map((studentRestriction) => ({
+    // Separate the results between non-legacy and legacy restrictions.
+    const nonLegacyRestrictions: StudentRestriction[] = [];
+    const legacyRestrictions: StudentRestriction[] = [];
+    studentRestrictions.forEach((studentRestriction) => {
+      const resultList = studentRestriction.restriction.isLegacy
+        ? legacyRestrictions
+        : nonLegacyRestrictions;
+      resultList.push(studentRestriction);
+    });
+    // Create the output result for non-legacy restrictions.
+    const results = nonLegacyRestrictions.map((studentRestriction) => ({
       code: studentRestriction.restriction.restrictionCode,
       type: studentRestriction.restriction.notificationType,
     }));
+    if (legacyRestrictions.length) {
+      const notificationOrder = Object.values(RestrictionNotificationType);
```

Comment:
Just a suggestion. Will it be better to use a map instead of the order how enums are declared?

```ts
 const notificationOrderMap = {
        [RestrictionNotificationType.Warning]: 1,
        [RestrictionNotificationType.Error]: 2,
        [RestrictionNotificationType.NoEffect]: 3,
      };
      // If any legacy restriction is present, create a generic LGCY restriction
      // with the highest notification type.
      studentRestrictions.sort(
        (a, b) =>
          notificationOrderMap[b.restriction.notificationType] -
          notificationOrderMap[a.restriction.notificationType],
      );
```

Reply:
I can see benefits in both ways.
Now that we do not have `NoEffect` anymore, maybe I can add a priority only to `Error` and `Warning` and the suggested code would be better and clear.

Reply:
Code update, please take a look.